### PR TITLE
Update contracts and testcases

### PR DESCRIPTION
**Pull request by SamAg19**
#19 Upgraded the contracts and fixed the errors in the test cases

### CODE REVIEW

**Comment by hrishikeshio** [comment]
Please mention relevant issue number in pr. this is for #19

File: build/IVoteManager.json [hunk 1]
```diff
@@ -0,0 +1 @@
...\":\"address\",\"name\":\"stakerAddress\",\"type\":\"address\"}],\"name\":\"reveal\",\"outputs\":[],\"stateMutability\":\"nonpayable\",\"type\":\"function\"}],\"devdoc\":{\"kind\":\"dev\",\"methods\":{},\"version\":1},\"userdoc\":{\"kind\":\"user\",\"methods\":{},\"version\":1}},\"settings\":{\"compilationTarget\":{\"IVoteManager.sol\":\"IVoteManager\"},\"evmVersion\":\"istanbul\",\"libraries\":{},\"metadata\":{\"bytecodeHash\":\"ipfs\"},\"optimizer\":{\"enabled\":false,\"runs\":200},\"remappings\":[]},\"sources\":{\"IVoteManager.sol\":{\"keccak256\":\"0xace1a3da1ee24d7d91cdb81b3b690c6ad5dc8f905a43897df0b05e24aad9e94e\",\"license\":\"GPL-3.0\",\"urls\":[\"bzz-raw://622e0888b695b9114178a4094260ba6062ee8582c1a9239b2c2df84c48cdb964\",\"dweb:/ipfs/QmPreGkGxVpXvxfhtoiUtZFoRoZXgoiaYwXqjLuxmsNQej\"]},\"lib/Structs.sol\":{\"keccak256\":\"0x305db0a22f904ccaca9c0aa6176ccaa0714a89f6962ee22005673ad572c7b215\",\"license\":\"GPL-3.0\",\"urls\":[\"bzz-raw://42ea703ef1c1d21577ce7c94a58c444cb20fa002dd1b3a112cca33c47c75b781\",\"dweb:/ipfs/QmQTD4EHuiqKs8u75gbtjaSHq8zqRxoiNuCC4vqz9H3Rcb\"]}},\"version\":1}","storageLayout":{"storage":[],"types":null},"userdoc":{"kind":"user","methods":{},"version":1}}
```
Comment (hrishikeshio) [inline build/IVoteManager.json:1]: Don't include build/ folder when sending a pr. Please remove this folder from your commit and update pr.

File: contracts/Bridge.sol [hunk 2]
```diff
@@ -1,4 +1,5 @@
-pragma solidity 0.5.10;
+// SPDX-License-Identifier: GPL-3.0
```
Comment (hrishikeshio) [inline contracts/Bridge.sol:1]: why was this line added?

File: truffle-config.js [hunk 3]
```diff
@@ -85,7 +85,7 @@ module.exports = {
                     return new HDWalletProvider(mnemonic, 'http://34.67.242.174:8545')
             },
             network_id: 5,
-            gas: 8000000,
+            gas: 4000000,
```
Comment (hrishikeshio) [inline truffle-config.js:88]: You are reverting your own change.

Make sure you merge the changes from the latest master in your branch to avoid such issues.

File: test/VoteManager.js [hunk 4]
```diff
@@ -35,13 +36,13 @@ contract('VoteManager', function (accounts) {
       // await stateManager.setEpoch(1)
       // await stateManager.setState(0)
       await functions.mineToNextEpoch()
-      await sch.transfer(accounts[3], 423000, { 'from': accounts[0]})
-      await sch.transfer(accounts[4], 19000, { 'from': accounts[0]})
-      await sch.approve(stakeManager.address, 420000, { 'from': accounts[3]})
-      await sch.approve(stakeManager.address, 19000, { 'from': accounts[4]})
+      await sch.transfer(accounts[3], new BN(423000).mul(new BN(10).pow(new BN('18'))), { 'from': accounts[0]})
```
Comment (hrishikeshio) [inline test/VoteManager.js:39]: any specific reason for this change?

avoid changes unrelated to the assigned issue.

the pr should only solve the assigned issue. unrelated changes should be avoided.

File: build/IVoteManager.json (hunk 1 above)
Comment (SamAg19) [reply build/IVoteManager.json:1]: Duly Noted. Sending a new pr with the changes mentioned.

File: contracts/Bridge.sol (hunk 2 above)
Comment (SamAg19) [reply contracts/Bridge.sol:1]: I was receiving the following warning if i do not add this line to the contracts:
SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.

Adding this line caused the warning to go away, it seems like the following warning comes up in solidity 6+

File: truffle-config.js (hunk 3 above)
Comment (SamAg19) [reply truffle-config.js:88]: Sorry about that.

File: test/VoteManager.js (hunk 4 above)
Comment (SamAg19) [reply test/VoteManager.js:39]: The previous code was causing the test cases in this file to fail. Making these changes made sure that the test cases were passing.

File: contracts/Core/WriterRole.sol [hunk 5]
```diff
@@ -1,44 +1,41 @@
...
+    
     event WriterAdded(address indexed account);
     event WriterRemoved(address indexed account);
-
-    Roles.Role private _writers;
-
+    
     constructor () internal {
-        _addWriter(msg.sender);
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
```
[the remark this answers is not in the document]
Comment (hrishikeshio) [reply contracts/Core/WriterRole.sol:16]: Please change MY_ROLE to more descriptive name. WRITER_ROLE

File: contracts/Core/IBlockManager.sol [hunk 6]
```diff
@@ -1,4 +1,4 @@
-pragma solidity 0.5.10;
+pragma solidity 0.6.11;
 pragma experimental ABIEncoderV2;
```
Comment (hrishikeshio) [inline contracts/Core/IBlockManager.sol:2]: I think this line isn't required anymore in solidity 0.6. please check docs and remove if it's okay to.

Comment (SamAg19) [reply contracts/Core/IBlockManager.sol:2]: I have to add this line. Gives me the following error:

TypeError: This type is only supported in ABIEncoderV2. Use "pragma experimental ABIEncoderV2;" to enable the feature.
    function getBlock(uint256 epoch) external view returns(Structs.Block memory _block);

Comment (hrishikeshio) [reply contracts/Core/IBlockManager.sol:2]: https://solidity.readthedocs.io/en/v0.6.12/layout-of-source-files.html?highlight=abi%20encoder%20v2#abiencoderv2
Seems it's not experimental but you still have to add this line.
